bsip-00XX Dynamic market fees.md

```diff
@@ -0,0 +1,183 @@
+
+Makers are so named because their orders make the liquidity in a market. Takers are the ones who remove this liquidity by matching makers' orders with their own.
+
+The maker-taker model encourages market liquidity by rewarding the makers of that liquidity with a fee discount. It also results in a tighter market spread due to the increased incentive for makers to outbid each other. The higher fee that the taker pays is usually offset by the better prices this tighter spread provides.
+
+**2) Supporting heavy volume traders.**
+Fees are charged and deduced on a per-trade basis. The bigger “trading volume” users have traded, the lower fee they get on subsequent trades.
+
+# General notes
+Each asset will have either usual market fee or a table with dynamic market fees that will represent the table above. 
+
+A switch **market_fee_type (REGULAR, DYNAMIC)** should clearly indicate, which market fee type is used.
+
+**Makers** are the ones who put up their orders first.
+
+And **takers** are the ones to put up their orders later in time than makers. 
+
+The following table 
+# Discussion
+
+We have had a discussion what's the best way to average out Trading Volume.
+
+There are a few ways of doing this: 
+
+- Use an **average volume since day the beginning of trading**.  
+Up for discussion is - what to take as a day 1. 1st trade of this asset? 1st trade minus 30 days?  Account open date? 
+This way we have a normal fair average volume if we figure out the fair beginning interval date.
+However, if a trader started trading this asset a few years ago he might be in a significant disadvantage.
+
+- **Average trading volume over the last 30 days - sliding window**.
+This way it is a bit clearer how to calculate. 
+However there might be extreme but real cases where this kind of calculation might not be fair: 
+a. when a person who usually trades high volumes throughout the year goes to vacation for 30 days. Suddenly all his discount is gone when he comes back
+b.   When another person comes in and makes 1 or 2 high volume trades while trading really low volumes before that. Suddenly this person might have a big discount. 
+30-day volume 
+This mehod gives enough information to figure out last 30-days traders habits with much lower memory requirements but slightly less precision than the 30-day average sliding window.
+Average trading volume over the last 365 days (a year) is a variation of this method.
+
+- **30-day volume method**
+This is quite similar to the 30-day average method and shares it's advantages and drawbacks.
+
+# Technical Specifications
+
+We suggest using average total volume for a particular asset. However, it's open for the discussion.
+
+## Statistics Storage
+
+There are two options for storing account statistics for trading:
+
+1. expand class account_statistics_object by adding a map of trade_statistics_object for each asset
+2. create a new type trade_statistics_object and a new index
```

Comment:
You'll need a statistic for every *base* asset (think USD) that you have traded.
Also stats should be reset after x weeks (4)

Reply:
Of course, there will be many of such objects per user (for each asset).
trade_statistics_objects will be stored in trade_statistics_index ordered by statistics object owner (user), asset.

Reply:
This can clutter the up the current state database significantly. Please indicate how garbage collection will work

Reply:
We plan to collect garbage while maintenance interval once a day in order to remove old data from the index.